x-pack/plugins/monitoring/server/lib/unlinked_deployments/has_unlinked_deployments.js

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import moment from 'moment';
+import { unlinkedDeploymentFilter } from './';
+
+export async function hasUnlinkedDeployments(req, indexPatterns) {
+  const indexPatternList = indexPatterns.reduce((list, patterns) => {
+    list.push(...patterns.split(','));
+    return list;
+  }, []);
+
+  // Get the params from the POST body for the request
+  const start = req.payload.timeRange.min;
+  const end = req.payload.timeRange.max;
+
+  const timeRangeFilter = {
+    range: {
+      timestamp: {
+        format: 'epoch_millis'
+      }
+    }
+  };
+
+  if (start) {
+    timeRangeFilter.range.timestamp.gte = moment.utc(start).valueOf();
+  }
+  if (end) {
+    timeRangeFilter.range.timestamp.lte = moment.utc(end).valueOf();
+  }
+
+  const msearch = indexPatternList.reduce((msearch, indexPattern) => {
+    msearch.push({
+      index: indexPattern,
+    });
+    msearch.push({
+      size: 1,
+      query: {
+        bool: {
+          filter: [ timeRangeFilter ],
+          must: [{
```

Comment:
And here, if `unlinkedDeploymentFilter` were an array, you could just add it to the `filter` array (not `must` since it has no impact on scoring and you want it to cache where possible!)

Reply:
Yup, fixed that up. Thanks!